specifyweb/specify/scoping.py

```diff
@@ -1,96 +1,134 @@
 from collections import namedtuple
+from typing import Tuple
+from django.db.models import Model
+from django.core.exceptions import ObjectDoesNotExist
 
 from . import models
 
-COLLECTION_SCOPE  = 0
-DISCIPLINE_SCOPE  = 1
-DIVISION_SCOPE    = 2
-INSTITUTION_SCOPE = 3
-GLOBAL_SCOPE      = 10
 
-class Scoping(namedtuple('Scoping', 'obj')):
+class ScopeType:
+    COLLECTION = 0
+    DISCIPLINE = 1
+    DIVISION = 2
+    INSTITUTION = 3
+    GLOBAL = 10
+
 
-    def __call__(self):
+class Scoping(namedtuple('Scoping', 'obj')):
+    def __call__(self) -> Tuple[int, Model]:
+        """
+            Returns the ScopeType and related Model instance of the 
+            hierarchical position the `obj` occupies. 
+            Tries and infers the scope based on the fields/relationships
+            on the model, and resolves the 'higher' scope before a more 
+            specific scope if applicable for the object
+        """
```

Comment:
(optional) I don't know python conventions well, but I commonly see it like this:

```suggestion
        """
        Returns the ScopeType and related Model instance of the 
        hierarchical position the `obj` occupies. 
        Tries and infers the scope based on the fields/relationships
        on the model, and resolves the 'higher' scope before a more 
        specific scope if applicable for the object
        """
```

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/43aee5b97e52edca55b58e0cb2b76773754b510f